Sources/ETCD/ETCDClient.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import GRPC
+import NIO
+import SwiftProtobuf
+
+
+public final class EtcdClient: Sendable {
+    private let host: String
+    private let port: Int
+    private var group: EventLoopGroup
+    private var connection: ClientConnection
+    private var client: Etcdserverpb_KVClient!
+
+    public init(host: String, port: Int, eventLoopGroup: EventLoopGroup) {
+        self.host = host
+        self.port = port
+        self.group = eventLoopGroup
+        self.connection = ClientConnection.insecure(group: self.group)
+            .connect(host: host, port: port)
+        self.client = Etcdserverpb_KVClient(channel: self.connection)
+    }
+
+    public func set(_ key: String, value: String) async throws {
+        var putRequest = Etcdserverpb_PutRequest()
+        putRequest.key = Data(key.utf8)
+        putRequest.value = Data(value.utf8)
+
+        let call = client.put(putRequest)
+        let response = try await call.response.get()
+    }
+
+    public func get(_ key: String) async throws -> String? {
```

Comment:
Similarly here. The function signature should end up as 
```suggestion
    public func get<Value: ETCDValue?(_ key: some ETCDKey, as valueType: Value.self) async throws -> Value? {
```

Reply:
For get and set, unsure how to change the putRequest.value. For set, should it be putRequest.value = value.asData? What would the return in get look like, since I wasn't able to return ETCDValue with the kv.value. E.g.
    ```
    public func set(_ key: String, value: some ETCDValue) async throws {
        putRequest.key = Data(key.utf8)
        putRequest.value = value.asData
    }

    public func get(_ key: String) async throws -> String? {
        ...

        guard let kv = response.kvs.first else {
            return nil
        }
        // return ETCDValue(data: kv.value, encoding: .utf8)
    }
    ```